### ISSUE TITLE
Add rule to prevent usage of `Object` type annotation

### ISSUE THREAD

**Issue by vitorbal** [zapier/eslint-plugin-zapier#42]
We almost never want to use `Object`, and should use `{}` instead. See also [the official docs](https://flow.org/en/docs/types/objects/#toc-object-type)

This would prevent bugs like this one: https://github.com/zapier/zapier/issues/18827#issuecomment-406423982

I think we can use [this eslint-plugin-flow rule](https://github.com/gajus/eslint-plugin-flowtype#no-weak-types)

**Comment by stevemolitor** [comment]
I assume the rule will allow the `$FlowFixMe` type?

**Comment by vitorbal** [comment]
yeah, `$FlowFixMe` would still work @stevemolitor! This is only to make sure devs aren't using `Object` and `Function` mistakenly!
